python/examples/core/2-handling-sparsity.py

```diff
@@ -0,0 +1,200 @@
+        representations for machine learning applications.
+
+        .. _rascaline: https://luthaf.fr/rascaline/latest/index.html
+
+        .. literalinclude:: radial-spectrum.py.example
+            :language: python
+
+The file contains a representation of two molecules called the radial spectrum. The atom
+:math:`i` is represented by the radial spectrum :math:`R_i^\alpha`, which is an
+expansion of the neighbor density :math:`\rho_i^\alpha(r)` on a set of radial basis
+functions :math:`f_n(r)`
+
+.. math::
+
+    R_i^\alpha(n) = \int f_n(r) \rho_i(r) dr
+
+The density :math:`\rho_i^\alpha(r)` associated with all neighbors of species
+:math:`\alpha` of the atom :math:`i` (each neighbor is replaced with a Gaussian function
+centered on the neighbor :math:`g(r_{ij})`) is defined as:
+
+.. math::
+
+    \rho_i^\alpha(r) = \sum_{j \in \text{ neighborhood of i }} g(r_{ij})
+        \delta_{\alpha_j,\alpha}
+
+
+The exact mathematical details above don't matter too much for this tutorial, the main
+point being that this representation treat atomic species as completely independent,
+effectively using the neighbor species :math:`\alpha` for `one-hot encoding`_.
+
+.. _one-hot encoding: https://en.wikipedia.org/wiki/One-hot
+
+.. py:currentmodule:: metatensor
+
+"""
+
+import ase
+import ase.visualize.plot
+import matplotlib.pyplot as plt
+import numpy as np
+
+import metatensor
+
+
+# %%
+#
+# We will work on the radial spectrum representation of three molecules in our system:
+# a carbone monoxyde, an oxygen molecule and a nitrogen molecule.
+
+atoms = ase.Atoms(
+    "COO2N2",
+    positions=[(0, 0, 0), (1.2, 0, 0), (0, 6, 0), (1.1, 6, 0), (6, 0, 0), (7.3, 0, 0)],
+)
+
+fig, ax = plt.subplots(figsize=(3, 3))
+ase.visualize.plot.plot_atoms(atoms, ax)
+ax.set_axis_off()
+plt.show()
+
+# %%
+#
+# Sparsity in ``TensorMap``
+# -------------------------
+#
+# The radial spectrum representation has two keys: ``central_species`` indicating the
+# species of the central atom (atom :math:`i` in the equations); and
+# ``neighbor_species`` indicating the species of the neighboring atoms (atom :math:`j`
+# in the equations)
+
+radial_spectrum = metatensor.load("radial-spectrum.npz")
+
+print(radial_spectrum)
+
+# %%
+#
+# This shows the first level of sparsity in ``TensorMap``: block sparsity.
+#
+# Out of all possible combinations of ``central_species`` and ``neighbor_species``, some
+# are missing such as ``central_species=7, neighbor_species=8``. This is because we are
+# using a spherical cutoff of 2.5 Å, and there are not oxygen atoms close enough to a
+# nitrogen. This means that all the corresponding radial spectrum coefficients
+# :math:`R_i^\alpha(n)` will be zero (since the neighbors density
+# :math:`\rho_i^\alpha(r)` is zero everywhere).
+#
+# Instead of wasting memory space to store all of these zeros explicitly, we don't
+# create the corresponding blocks from the get go and save a lot of memory!
```

Comment:
```suggestion
# Instead of wasting memory space by storing all of these zeros explicitly, we simply
# avoid creating the corresponding blocks from the get-go and save a lot of memory!
```